dev-guide/cluster-version-operator/dev/clusteroperator.md

```diff
@@ -207,3 +207,49 @@ Conditions determine when the CVO considers certain actions complete, the follow
 [2] Upgrade will not proceed with upgrading components in the next runlevel until the previous runlevel completes.
 
 See also: https://github.com/openshift/cluster-version-operator/blob/a5f5007c17cc14281c558ea363518dcc5b6675c7/pkg/cvo/internal/operatorstatus.go#L176-L189
+
+### Related Objects
+
+Related objects are the fundamental set of objects related to your operator.
+As mentioned previously, the insights operator and must-gather depend on related images to collect relevant diagnostic information about each ClusterOperator.
+
+In addition to specifying related objects statically in your operator's manifests, you should also implement logic in your operator to keep related objects up-to-date.
+This ensures that another client cannot permanently change your related objects, and it enables the operator to change related objects dynamically based on operator configuration or cluster state. 
+
+Related objects are identified by their `group`, `resource`, and `name` (and `namespace` for namespace-scoped resources).
+
+To declare a namespace as a related object:
+```yaml
+# NOTE: "" is the core group
+- group: ""
+  resource: "namespaces"
+  name: "openshift-component"
+```
+
+To declare a configmap as a related object:
+```yaml
+- group: ""
+  resource: "configmaps"
+  namespace: "openshift-component"
```

Comment:
might be worth making this a shared namespace like `openshift-config` or `kube-system`.  Because inspection logic in `oc` (and probably also Insights?) collect [a bunch of resources][1] inside a referenced namespace, including all ConfigMaps.  It doesn't hurt to back that up with an explicit declaration, but there's some risk that folks who see an explicit declaration will forget about the implicit namespace inclusions, and be confused when considering a separate ClusterOperator that does not make the explicit declaration.  I'm personally fine with an org-scoped policy like "ClusterOperator should not rely on implicit namespace resource gathering", but that's not where we are today, and it would take some work for existing components to get us there, even if we did get consensus agreement on that being a good place to go.

[1]: https://github.com/openshift/oc/blob/d652ca32a5e1c95be16f802ec52c2e04857327c2/pkg/cli/admin/inspect/namespace.go#L19-L35

Reply:
Ah! I was actually wondering if there was some magic "collect all these things from any declared `namespace` related object".

I could also change the group/resource to one of the not-default-collected things. Or better yet, just do both.

Reply:
Ok, updated to split this example in two.